zrml/prediction-markets/src/lib.rs

```diff
@@ -2035,7 +2038,11 @@ mod pallet {
         ) -> DispatchResult {
             ensure!(market.report.is_some(), Error::<T>::MarketNotReported);
             Self::ensure_outcome_matches_market_type(market, outcome)?;
-            Self::ensure_can_not_dispute_the_same_outcome(disputes, outcome)?;
+            Self::ensure_can_not_dispute_the_same_outcome(
+                disputes,
+                &market.report.as_ref().unwrap(),
```

Comment:
We generally avoid code that can panic.
```suggestion
                (&market.report.as_ref()).ok_or(Error::<T>::MarketIsNotReported)?
```

Reply:
Understood. Doesn't `ensure!(market.report.is_some(), Error::<T>::MarketNotReported);` guarantee that `unwrap()` doesn't fail, though?

Reply:
Yes, it does. It's just a philosophy to avoid code that can potentially panic, even if it is clear that it can not occur in the current state of the code.

Reply:
Ok, got it!